### DISCUSSION TITLE
Auto3DSeg: CT modality; converting to Cormack units

### DISCUSSION THREAD

**Discussion by paulwrightwork** [Project-MONAI/MONAI#7454] [Q&A]
My general question is what setting the Auto3DSeg modality to CT does.

My specific question is about converting my CT images to Cormack units as described [here](https://doi.org/10.1016/j.neuroimage.2012.03.020) and implemented [here](https://github.com/neurolabusc/Clinical/blob/master/clinical_h2c.m). I am considering doing this to make the model more sensitive to the range of values where the lesion I want to segment usually occurs. My concern is that this may conflict with whatever transforms Auto3DSeg uses for CT modality. In short, converting Hounsfield to Cormack units sets the minimum voxel value to zero and expands the dynamic range of a middle range of values (e.g. -100 to +100 HU).

Please advise whether transforming the images before using Auto3DSeg will cause a problem. If applicable, perhaps there is a solution that involves incorporating the Hounsfield to Cormack transformation in the set of MONAI transforms.

Many thanks.

**Comment by paulwrightwork** [comment]
Can someone comment on this please?

Should I set the modality to MR if I rescale my CTs as described?

Does Augo3DSeg already include a transform like this for CTs so it won't have any effect or, worse, will conflict?

Thank you!

**Reply by KumoLiu** [reply]
Hi @pwrightkcl, there is a placeholder transform for intensity for each algorithm in Auto3dSeg. It will select the proper transform by determining the modality.
https://github.com/Project-MONAI/research-contributions/blob/602bbe5b95ef29d56d1a933d5bd58af3ec144f6a/auto3dseg/algorithm_templates/dints/scripts/algo.py#L188
https://github.com/Project-MONAI/research-contributions/blob/602bbe5b95ef29d56d1a933d5bd58af3ec144f6a/auto3dseg/algorithm_templates/segresnet/scripts/algo.py#L135
https://github.com/Project-MONAI/research-contributions/blob/602bbe5b95ef29d56d1a933d5bd58af3ec144f6a/auto3dseg/algorithm_templates/segresnet2d/scripts/algo.py#L157
https://github.com/Project-MONAI/research-contributions/blob/602bbe5b95ef29d56d1a933d5bd58af3ec144f6a/auto3dseg/algorithm_templates/swinunetr/scripts/algo.py#L196

Hope it helps, thanks.

**Reply by paulwrightwork** [reply]
It helps a bit. I'll have to do more digging to find out what that does downstream. Thanks.
